airbyte-config/config-models/src/main/resources/AirbyteTypes.yaml

```diff
@@ -0,0 +1,13 @@
+definitions:
+  TimestampWithTimezone:
+    type: String
+    description: ISO8601 encoded
```

Comment:
why are these all the same `8601`? I would assume they should be different for validation purposes in JSONSchema.

Reply:
we could also accompany these with `pattern` directives to make it really clear what we expect https://json-schema.org/understanding-json-schema/reference/regular_expressions.html

Reply:
for sure!

the point i want to get across is that there are 2 concepts here: 
1) airbyte's notion of what the type is 
2) how that type is actually serialized. 

every single one of the temporal types that airbyte wants to support (timestamp, date, time, with tz, without tz) _can_ be represented using the same serialization protocol (timestamp is a superset of all of these). as long as we know that this serialized timestamp should just be treated like a date, for example, we can get everything we want. if the tool we use does have a tighter type for us to use, we certainly can (and should use it), but it is not a requirement. 

this realization was important for me to realize that even though protobuf does _not_ let you add custom scalar types AND has a less granular type system than we want, we are _still_ able to build the richer type system we want on top of that, because using the existing scalars we can serialize anything we need to. 

the same is true for JSONSchema as well. we don't need to forgo using the `type` field in JSONSchema is we don't want to. continuing to use the `type` field allows us to stay more inline with the JSONSchema ecosystem.

Reply:
it sounds like there was an in-person convo so I might be missing some context here - what does this look like in practice? I.e. in a catalog, where we currently have a `schema` field (which is a JSONSchema) - would that turn into a `json_schema` (used purely for... data validation, maybe?) + `more_specific_airbyte_schema`?

generally I agree that forgoing native jsonschema `type` is _mostly_ orthogonal to our type system - the edge case that using `string` everywhere solves is infinity/nan, which aren't valid json keywords. That's not an issue in protobuf though, it's purely a consequence of using json-based transport